src/Containers/MiddleCanvas/Template.tsx

```diff
@@ -47,20 +33,18 @@ export const Template: React.FC<TemplateProps> = ({
     const onDrag = (result: DropResult): void => {
         const { source, destination } = result;
     
-        if(!destination) {
-            return;
-        }
-
-        const reorderedList = reorder(items, source.index, destination.index);
+        if(!destination) return;
+    
+        const reorderedList = reorderList(items, source.index, destination.index);
         setItems(reorderedList);
     };
 
     /**
      * Returns a header component based on component type
-     * @param {componentType} componentType - type of component (like a table)
+     * @param {string | undefined} componentType - type of component (like a table)
      * @param {string[][]} labels - labels needed to be mapped in each individual component
      */
-    const conditionalHeaderComponent = (componentType: string, labels: string[][]): React.ReactElement | undefined => {
+    const conditionalHeaderComponent: getHeaderComponentType = (componentType, labels) => {
```

Comment:
For readability let's type the function rather than giving it a type. Ex: 
const conditionalHeaderComponent = (componentType: string | undefined, labels: string[ ][ ]): Reac.ReactElement =>